pages/add-habit.tsx

```diff
@@ -0,0 +1,208 @@
+import { css } from '@emotion/react';
+import Image from 'next/image';
+import Button from '../components/Button';
+import Link from '../src/components/Link';
+import habitsApi from '../src/api/habits';
+import { FormEvent, useState } from 'react';
+import { NextRouter, useRouter } from 'next/router';
+
+type FormType = {
+  name?: string;
+  description?: string;
+  target?: number;
+  unit?: string;
+  frequency: number;
+  frequencyUnit: string;
+};
+
+const defaultFormValues = {
+  name: 'Drink water',
+  target: 1,
+  frequency: 1,
+  frequencyUnit: 'daily'
+};
+
+const styles = {
+  screen: css({
+    backgroundColor: '#292735',
+    minHeight: '100vh',
+    minWidth: '100vw',
+    color: '#fff'
+  }),
+
+  container: css({
+    display: 'flex',
+    flexDirection: 'column',
+    width: '80%',
+    margin: '0 auto',
+    padding: '2em',
+    alignItems: 'center'
+  }),
+
+  form: css({
+    display: 'flex',
+    flexDirection: 'column',
+    maxWidth: '800px'
+  }),
+
+  inputField: css({
+    margin: '1em 0',
+    padding: '0.7em',
+    backgroundColor: '#220000',
+    color: '#fff'
+  }),
+
+  horizontalFlexContainer: css({
+    display: 'flex',
+    alignItems: 'center'
+  }),
+  verticalFlexContainer: css({
+    display: 'flex',
+    flexDirection: 'column',
+    width: '33%'
+  }),
+
+  buttonsContainer: css({
+    display: 'flex',
+    justifyContent: 'space-between',
+    marginTop: '1em'
+  }),
+  button: css({
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    width: '150px',
+    borderRadius: '1em'
+  }),
+  buttonText: css({
+    margin: '0 0.7em'
+  })
+};
+
+const handleSubmit = (event: FormEvent, router: NextRouter) => {
+  event.preventDefault();
+  habitsApi.addHabit().then((message) => {
+    console.log(message);
+    router.push('/');
+  });
+};
+
+const AddHabit = () => {
+  const [formData, setFormData] = useState<FormType>(defaultFormValues);
+
+  const router = useRouter();
+
+  const handleInputChange = (
+    event: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>
+  ) => {
+    const { name, value } = event.target;
+    // const value = target?.type === 'checkbox' ? target.checked : target.value;
+
+    setFormData({ ...formData, [name]: value });
+    console.log(`${name} changed to ${value}`);
+  };
+
+  return (
+    <div css={styles.screen}>
+      <div css={styles.container}>
+        <h2>Add a habit</h2>
+        <form css={styles.form} onSubmit={(e) => handleSubmit(e, router)}>
```

Comment:
It may make more sense to put the handler inside the component - this would mean that you do not need to pass through the router, and be able to reduce the form to:
`onSubmit={(e) => handleSubmit(e)}`, which becomes `onSubmit={handleSubmit}`

Reply:
Smart. Done that now :)